diff -Naur patchtemp/lib.before/lua-5.1.5/meson.build patchtemp/lib.after/lua-5.1.5/meson.build
--- patchtemp/lib.before/lua-5.1.5/meson.build	1970-01-01 01:00:00.000000000 +0100
+++ patchtemp/lib.after/lua-5.1.5/meson.build	2021-01-03 00:44:30.393167369 +0100
@@ -0,0 +1,37 @@
+project('tpt-libs-win64-lua5.1', 'c')
+
+static_library(
+	'lua5.1',
+	sources: files(
+		'src/lapi.c',
+		'src/lauxlib.c',
+		'src/lbaselib.c',
+		'src/lcode.c',
+		'src/ldblib.c',
+		'src/ldebug.c',
+		'src/ldo.c',
+		'src/ldump.c',
+		'src/lfunc.c',
+		'src/lgc.c',
+		'src/linit.c',
+		'src/liolib.c',
+		'src/llex.c',
+		'src/lmathlib.c',
+		'src/lmem.c',
+		'src/loadlib.c',
+		'src/lobject.c',
+		'src/lopcodes.c',
+		'src/loslib.c',
+		'src/lparser.c',
+		'src/lstate.c',
+		'src/lstring.c',
+		'src/lstrlib.c',
+		'src/ltable.c',
+		'src/ltablib.c',
+		'src/ltm.c',
+		'src/lundump.c',
+		'src/lvm.c',
+		'src/lzio.c',
+	),
+	c_args: [ '/TP' ] # msvc "it's actually all C++" flag
+)
