diff -Naur curl-7.86.0.old/CMakeLists.txt curl-7.86.0.new/CMakeLists.txt
--- curl-7.86.0.old/CMakeLists.txt	2023-10-02 22:24:12.603386800 +0200
+++ curl-7.86.0.new/CMakeLists.txt	2023-10-02 22:24:41.662508100 +0200
@@ -1308,6 +1308,7 @@
   if(USE_WIN32_CRYPTO OR USE_SCHANNEL)
     list(APPEND CURL_LIBS "advapi32" "crypt32")
   endif()
+  list(APPEND CURL_LIBS "ws2_32")
 
   # Matching logic used for Curl_win32_random()
   if(MINGW)
