diff --strip-trailing-cr -Naur libpng-1.6.49.old/CMakeLists.txt libpng-1.6.49.new/CMakeLists.txt
--- libpng-1.6.49.old/CMakeLists.txt	2025-06-21 16:01:05.696480646 +0200
+++ libpng-1.6.49.new/CMakeLists.txt	2025-06-21 16:01:25.718693409 +0200
@@ -350,11 +350,6 @@
 
 else(PNG_HARDWARE_OPTIMIZATIONS)
 
-  # Set definitions and sources for ARM.
-  if(PNG_TARGET_ARCHITECTURE MATCHES "^(arm|aarch)")
-    add_definitions(-DPNG_ARM_NEON_OPT=0)
-  endif()
-
   # Set definitions and sources for PowerPC.
   if(PNG_TARGET_ARCHITECTURE MATCHES "^(powerpc|ppc64)")
     add_definitions(-DPNG_POWERPC_VSX_OPT=0)
@@ -382,6 +377,9 @@
 
 endif(PNG_HARDWARE_OPTIMIZATIONS)
 
+# Set definitions and sources for ARM.
+add_definitions(-DPNG_ARM_NEON_OPT=0)
+
 option(ld-version-script "Enable linker version script" ON)
 if(ld-version-script AND NOT (ANDROID OR APPLE))
   # Check if LD supports linker scripts.
