diff -Naur lua-5.1.5.old/src/lauxlib.h lua-5.1.5.new/src/lauxlib.h
--- lua-5.1.5.old/src/lauxlib.h	2022-08-24 11:12:21.681839983 +0200
+++ lua-5.1.5.new/src/lauxlib.h	2022-08-24 11:12:56.196134658 +0200
@@ -12,6 +12,11 @@
 #include <stddef.h>
 #include <stdio.h>
 
+#ifdef __cplusplus
+extern "C"
+{
+#endif
+
 #include "lua.h"
 
 
@@ -169,6 +174,10 @@
 
 #define luaL_reg	luaL_Reg
 
+#ifdef __cplusplus
+}
+#endif
+
 #endif
 
 
diff -Naur lua-5.1.5.old/src/lua.h lua-5.1.5.new/src/lua.h
--- lua-5.1.5.old/src/lua.h	2022-08-24 11:12:21.681839983 +0200
+++ lua-5.1.5.new/src/lua.h	2022-08-24 11:12:21.685173409 +0200
@@ -12,6 +12,10 @@
 #include <stdarg.h>
 #include <stddef.h>
 
+#ifdef __cplusplus
+extern "C"
+{
+#endif
 
 #include "luaconf.h"
 
@@ -384,5 +388,8 @@
 * SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
 ******************************************************************************/
 
+#ifdef __cplusplus
+}
+#endif
 
 #endif
diff -Naur lua-5.1.5.old/src/lualib.h lua-5.1.5.new/src/lualib.h
--- lua-5.1.5.old/src/lualib.h	2022-08-24 11:12:21.681839983 +0200
+++ lua-5.1.5.new/src/lualib.h	2022-08-24 11:12:56.279470312 +0200
@@ -8,6 +8,11 @@
 #ifndef lualib_h
 #define lualib_h
 
+#ifdef __cplusplus
+extern "C"
+{
+#endif
+
 #include "lua.h"
 
 
@@ -49,5 +54,8 @@
 #define lua_assert(x)	((void)0)
 #endif
 
+#ifdef __cplusplus
+}
+#endif
 
 #endif
