diff -Naur SDL2-2.24.2.old/CMakeLists.txt SDL2-2.24.2.new/CMakeLists.txt
--- SDL2-2.24.2.old/CMakeLists.txt	2022-11-08 08:21:11.483847256 +0100
+++ SDL2-2.24.2.new/CMakeLists.txt	2022-11-08 08:22:34.466302697 +0100
@@ -539,7 +539,6 @@
   ${SDL2_SOURCE_DIR}/src/events/*.c
   ${SDL2_SOURCE_DIR}/src/file/*.c
   ${SDL2_SOURCE_DIR}/src/joystick/*.c
-  ${SDL2_SOURCE_DIR}/src/haptic/*.c
   ${SDL2_SOURCE_DIR}/src/hidapi/*.c
   ${SDL2_SOURCE_DIR}/src/libm/*.c
   ${SDL2_SOURCE_DIR}/src/locale/*.c
@@ -1053,6 +1052,8 @@
     # Haptic requires some private functions from the joystick subsystem.
     message_error("SDL_HAPTIC requires SDL_JOYSTICK, which is not enabled")
   endif()
+  file(GLOB JOYSTICK_SOURCES ${SDL2_SOURCE_DIR}/src/haptic/*.c)
+  set(SOURCE_FILES ${SOURCE_FILES} ${JOYSTICK_SOURCES})
 endif()
 
 
