diff -Naur patchtemp/lib.before/LuaJIT-2.1.0-beta3/src/Makefile patchtemp/lib.after/LuaJIT-2.1.0-beta3/src/Makefile
--- patchtemp/lib.before/LuaJIT-2.1.0-beta3/src/Makefile	2017-05-01 21:05:00.000000000 +0200
+++ patchtemp/lib.after/LuaJIT-2.1.0-beta3/src/Makefile	2021-02-23 08:55:46.911219000 +0100
@@ -71,10 +71,10 @@
 # as dynamic mode.
 #
 # Mixed mode creates a static + dynamic library and a statically linked luajit.
-BUILDMODE= mixed
+#BUILDMODE= mixed
 #
 # Static mode creates a static library and a statically linked luajit.
-#BUILDMODE= static
+BUILDMODE= static
 #
 # Dynamic mode creates a dynamic library and a dynamically linked luajit.
 # Note: this executable will only run when the library is installed!
