diff -Naur patchtemp/lib.before/lua-5.1.5/src/llex.c patchtemp/lib.after/lua-5.1.5/src/llex.c
--- patchtemp/lib.before/lua-5.1.5/src/llex.c	2009-11-23 15:58:22.000000000 +0100
+++ patchtemp/lib.after/lua-5.1.5/src/llex.c	2022-04-12 07:45:59.836725258 +0200
@@ -178,7 +178,7 @@
 
 static void trydecpoint (LexState *ls, SemInfo *seminfo) {
   /* format error: try to update decimal point separator */
-  struct lconv *cv = localeconv();
+  struct lconv *cv = NULL;
   char old = ls->decpoint;
   ls->decpoint = (cv ? cv->decimal_point[0] : '.');
   buffreplace(ls, old, ls->decpoint);  /* try updated decimal separator */
diff -Naur patchtemp/lib.before/lua-5.1.5/src/luaconf.h patchtemp/lib.after/lua-5.1.5/src/luaconf.h
--- patchtemp/lib.before/lua-5.1.5/src/luaconf.h	2008-02-11 17:25:08.000000000 +0100
+++ patchtemp/lib.after/lua-5.1.5/src/luaconf.h	2022-04-12 07:45:23.902636465 +0200
@@ -36,7 +36,7 @@
 #if defined(LUA_USE_LINUX)
 #define LUA_USE_POSIX
 #define LUA_USE_DLOPEN		/* needs an extra library: -ldl */
-#define LUA_USE_READLINE	/* needs some extra libraries */
+// #define LUA_USE_READLINE	/* needs some extra libraries */
 #endif
 
 #if defined(LUA_USE_MACOSX)
diff -Naur patchtemp/lib.before/lua-5.1.5/src/Makefile patchtemp/lib.after/lua-5.1.5/src/Makefile
--- patchtemp/lib.before/lua-5.1.5/src/Makefile	2012-02-13 21:41:22.000000000 +0100
+++ patchtemp/lib.after/lua-5.1.5/src/Makefile	2022-04-12 07:45:23.869302429 +0200
@@ -47,15 +47,18 @@
 
 a:	$(ALL_A)
 
+%.o: %.c
+	$(CC) $(CFLAGS) -c $< -o $@
+
 $(LUA_A): $(CORE_O) $(LIB_O)
 	$(AR) $@ $(CORE_O) $(LIB_O)	# DLL needs all object files
 	$(RANLIB) $@
 
 $(LUA_T): $(LUA_O) $(LUA_A)
-	$(CC) -o $@ $(MYLDFLAGS) $(LUA_O) $(LUA_A) $(LIBS)
+	$(CC) $(CFLAGS) -o $@ $(MYLDFLAGS) $(LUA_O) $(LUA_A) $(LIBS)
 
 $(LUAC_T): $(LUAC_O) $(LUA_A)
-	$(CC) -o $@ $(MYLDFLAGS) $(LUAC_O) $(LUA_A) $(LIBS)
+	$(CC) $(CFLAGS) -o $@ $(MYLDFLAGS) $(LUAC_O) $(LUA_A) $(LIBS)
 
 clean:
 	$(RM) $(ALL_T) $(ALL_O)
@@ -90,16 +93,16 @@
 	$(MAKE) all MYCFLAGS="-DLUA_USE_POSIX -DLUA_USE_DLOPEN" MYLIBS="-Wl,-E"
 
 freebsd:
-	$(MAKE) all MYCFLAGS="-DLUA_USE_LINUX" MYLIBS="-Wl,-E -lreadline"
+	$(MAKE) all MYCFLAGS="-DLUA_USE_LINUX" MYLIBS="-Wl,-E"
 
 generic:
 	$(MAKE) all MYCFLAGS=
 
 linux:
-	$(MAKE) all MYCFLAGS=-DLUA_USE_LINUX MYLIBS="-Wl,-E -ldl -lreadline -lhistory -lncurses"
+	$(MAKE) all MYCFLAGS=-DLUA_USE_LINUX MYLIBS="-Wl,-E -ldl"
 
 macosx:
-	$(MAKE) all MYCFLAGS=-DLUA_USE_LINUX MYLIBS="-lreadline"
+	$(MAKE) all MYCFLAGS=-DLUA_USE_LINUX MYLIBS=""
 # use this on Mac OS X 10.3-
 #	$(MAKE) all MYCFLAGS=-DLUA_USE_MACOSX
 
