diff -Naur patchtemp/lib.before/lua-5.2.4/src/llex.c patchtemp/lib.after/lua-5.2.4/src/llex.c
--- patchtemp/lib.before/lua-5.2.4/src/llex.c	2015-02-09 18:56:34.000000000 +0100
+++ patchtemp/lib.after/lua-5.2.4/src/llex.c	2022-04-12 07:44:59.132113575 +0200
@@ -202,7 +202,7 @@
 
 
 #if !defined(getlocaledecpoint)
-#define getlocaledecpoint()	(localeconv()->decimal_point[0])
+#define getlocaledecpoint()	'.'
 #endif
 
 
diff -Naur patchtemp/lib.before/lua-5.2.4/src/luaconf.h patchtemp/lib.after/lua-5.2.4/src/luaconf.h
--- patchtemp/lib.before/lua-5.2.4/src/luaconf.h	2013-11-21 18:26:16.000000000 +0100
+++ patchtemp/lib.after/lua-5.2.4/src/luaconf.h	2022-04-12 07:44:08.124364283 +0200
@@ -43,7 +43,7 @@
 #if defined(LUA_USE_LINUX)
 #define LUA_USE_POSIX
 #define LUA_USE_DLOPEN		/* needs an extra library: -ldl */
-#define LUA_USE_READLINE	/* needs some extra libraries */
+// #define LUA_USE_READLINE	/* needs some extra libraries */
 #define LUA_USE_STRTODHEX	/* assume 'strtod' handles hex formats */
 #define LUA_USE_AFORMAT		/* assume 'printf' handles 'aA' specifiers */
 #define LUA_USE_LONGLONG	/* assume support for long long */
@@ -52,7 +52,7 @@
 #if defined(LUA_USE_MACOSX)
 #define LUA_USE_POSIX
 #define LUA_USE_DLOPEN		/* does not need -ldl */
-#define LUA_USE_READLINE	/* needs an extra library: -lreadline */
+// #define LUA_USE_READLINE	/* needs an extra library: -lreadline */
 #define LUA_USE_STRTODHEX	/* assume 'strtod' handles hex formats */
 #define LUA_USE_AFORMAT		/* assume 'printf' handles 'aA' specifiers */
 #define LUA_USE_LONGLONG	/* assume support for long long */
diff -Naur patchtemp/lib.before/lua-5.2.4/src/Makefile patchtemp/lib.after/lua-5.2.4/src/Makefile
--- patchtemp/lib.before/lua-5.2.4/src/Makefile	2013-11-11 12:45:49.000000000 +0100
+++ patchtemp/lib.after/lua-5.2.4/src/Makefile	2022-04-12 07:44:08.031028961 +0200
@@ -55,15 +55,18 @@
 
 a:	$(ALL_A)
 
+%.o: %.c
+	$(CC) $(CFLAGS) -c $< -o $@
+
 $(LUA_A): $(BASE_O)
 	$(AR) $@ $(BASE_O)
 	$(RANLIB) $@
 
 $(LUA_T): $(LUA_O) $(LUA_A)
-	$(CC) -o $@ $(LDFLAGS) $(LUA_O) $(LUA_A) $(LIBS)
+	$(CC) $(CFLAGS) -o $@ $(LDFLAGS) $(LUA_O) $(LUA_A) $(LIBS)
 
 $(LUAC_T): $(LUAC_O) $(LUA_A)
-	$(CC) -o $@ $(LDFLAGS) $(LUAC_O) $(LUA_A) $(LIBS)
+	$(CC) $(CFLAGS) -o $@ $(LDFLAGS) $(LUAC_O) $(LUA_A) $(LIBS)
 
 clean:
 	$(RM) $(ALL_T) $(ALL_O)
@@ -98,15 +101,15 @@
 	$(MAKE) $(ALL) SYSCFLAGS="-DLUA_USE_POSIX -DLUA_USE_DLOPEN" SYSLIBS="-Wl,-E"
 
 freebsd:
-	$(MAKE) $(ALL) SYSCFLAGS="-DLUA_USE_LINUX" SYSLIBS="-Wl,-E -lreadline"
+	$(MAKE) $(ALL) SYSCFLAGS="-DLUA_USE_LINUX" SYSLIBS="-Wl,-E"
 
 generic: $(ALL)
 
 linux:
-	$(MAKE) $(ALL) SYSCFLAGS="-DLUA_USE_LINUX" SYSLIBS="-Wl,-E -ldl -lreadline"
+	$(MAKE) $(ALL) SYSCFLAGS="-DLUA_USE_LINUX" SYSLIBS="-Wl,-E -ldl"
 
 macosx:
-	$(MAKE) $(ALL) SYSCFLAGS="-DLUA_USE_MACOSX" SYSLIBS="-lreadline" CC=cc
+	$(MAKE) $(ALL) SYSCFLAGS="-DLUA_USE_MACOSX" SYSLIBS=""
 
 mingw:
 	$(MAKE) "LUA_A=lua52.dll" "LUA_T=lua.exe" \
