diff -Naur patchtemp/lib.before/SDL2-2.0.20/src/dynapi/SDL_dynapi.h patchtemp/lib.after/SDL2-2.0.20/src/dynapi/SDL_dynapi.h
--- patchtemp/lib.before/SDL2-2.0.20/src/dynapi/SDL_dynapi.h	2019-07-25 06:32:36.000000000 +0200
+++ patchtemp/lib.after/SDL2-2.0.20/src/dynapi/SDL_dynapi.h	2021-10-21 15:49:53.823122200 +0200
@@ -59,7 +59,7 @@
 
 /* everyone else. This is where we turn on the API if nothing forced it off. */
 #ifndef SDL_DYNAMIC_API
-#define SDL_DYNAMIC_API 1
+// #define SDL_DYNAMIC_API 1
 #endif
 
 #endif
