diff -Naur SDL2-2.0.10/CMakeLists.txt SDL2-2.0.10-nd/CMakeLists.txt
--- SDL2-2.0.10/CMakeLists.txt	2019-07-25 06:32:36.000000000 +0200
+++ SDL2-2.0.10-nd/CMakeLists.txt	2020-07-30 15:36:57.213279300 +0200
@@ -1335,7 +1335,7 @@
   endif()
 
   # Libraries for Win32 native and MinGW
-  list(APPEND EXTRA_LIBS user32 gdi32 winmm imm32 ole32 oleaut32 version uuid advapi32 setupapi shell32)
+  list(APPEND EXTRA_LIBS vcruntime user32 gdi32 winmm imm32 ole32 oleaut32 version uuid advapi32 setupapi shell32)
 
   # TODO: in configure.ac the check for timers is set on
   # cygwin | mingw32* - does this include mingw32CE?
diff -Naur SDL2-2.0.10/src/dynapi/SDL_dynapi.h SDL2-2.0.10-nd/src/dynapi/SDL_dynapi.h
--- SDL2-2.0.10/src/dynapi/SDL_dynapi.h	2019-07-25 06:32:36.000000000 +0200
+++ SDL2-2.0.10-nd/src/dynapi/SDL_dynapi.h	2020-07-30 15:35:33.966525200 +0200
@@ -59,7 +59,7 @@
 
 /* everyone else. This is where we turn on the API if nothing forced it off. */
 #ifndef SDL_DYNAMIC_API
-#define SDL_DYNAMIC_API 1
+// #define SDL_DYNAMIC_API 1
 #endif
 
 #endif
