diff -Naur SDL2-2.0.10/configure SDL2-2.0.10-ni/configure
--- SDL2-2.0.10/configure	2019-07-25 06:32:36.000000000 +0200
+++ SDL2-2.0.10-ni/configure	2020-06-05 12:31:59.039473075 +0200
@@ -17083,7 +17083,7 @@
   ac_fn_c_check_func "$LINENO" "iconv" "ac_cv_func_iconv"
 if test "x$ac_cv_func_iconv" = xyes; then :
   cat >>confdefs.h <<_ACEOF
-#define HAVE_ICONV 1
+//#define HAVE_ICONV 1
 _ACEOF
 
 fi
