diff -Naur LuaJIT-2.1.0-beta3/src/msvcbuild.bat LuaJIT-2.1.0-beta3-pn/src/msvcbuild.bat
--- LuaJIT-2.1.0-beta3/src/msvcbuild.bat	2017-05-01 21:05:00.000000000 +0200
+++ LuaJIT-2.1.0-beta3-pn/src/msvcbuild.bat	2020-07-30 12:17:39.151458800 +0200
@@ -21,8 +21,8 @@
 @set DASMDIR=..\dynasm
 @set DASM=%DASMDIR%\dynasm.lua
 @set DASC=vm_x86.dasc
-@set LJDLLNAME=lua51.dll
-@set LJLIBNAME=lua51.lib
+@set LJDLLNAME=luajit21.dll
+@set LJLIBNAME=luajit21.lib
 @set ALL_LIB=lib_base.c lib_math.c lib_bit.c lib_string.c lib_table.c lib_io.c lib_os.c lib_package.c lib_debug.c lib_jit.c lib_ffi.c
 
 %LJCOMPILE% host\minilua.c
